bin/tritond

```diff
@@ -0,0 +1,313 @@
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--verbose', '-v',
+        dest='verbose',
+        action='append_const',
+        const=True,
+        default=list())
+    parser.add_argument(
+        '--log-path', '-l',
+        dest='log_path',
+        action='store',
+        default=None,
+        help="Log files of messages sent will be written if this path is set")
+    parser.add_argument(
+        '--skip-kinesis',
+        dest='skip_kinesis',
+        action='store_true',
+        default=False,
+        help="will skip sends to Kinesis; for debug purposess")
+
+    options = parser.parse_args()
+
+    setup_logging(options)
+
+    continue_running = [True]
+
+    def handle_sigterm(signum, frame):
+        log.info("Exiting")
+        continue_running[0] = False
+
+    signal.signal(signal.SIGTERM, handle_sigterm)
+    signal.signal(signal.SIGINT, handle_sigterm)
+
+    if options.log_path:
+        try:
+            os.makedirs(options.log_path)
+        except OSError:
+            pass
+
+    zmq_context = zmq.Context()
+    poller = zmq.Poller()
+
+    collect_host = "{}:{}".format(*config.get_zmq_config())
+    log.info("Initializing collector port %s", collect_host)
+    collector_sock = zmq_context.socket(zmq.PULL)
+    collector_sock.bind("tcp://%s" % collect_host)
+    poller.register(collector_sock, zmq.POLLIN)
+
+    last_write = time.time()
+    waiting_messages = defaultdict(list)
+    log_files = dict()
+
+    log.info("Starting IO Loop")
+    while continue_running[0]:
+        log.debug("Poll")
+
+        try:
+            ready = dict(poller.poll(POLL_LOOP_TIMEOUT_MS))
+        except (KeyboardInterrupt, SystemExit):
+            continue_running[0] = False
+            break
+        except zmq.ZMQError, e:
+            if e.errno == errno.EINTR:
+                # If this is from a SIGTERM, we have a handler for that and the
+                # loop should exit gracefull.
+                continue
+            else:
+                raise
+
+        log.debug("Poller returned: %r", ready)
+
+        if collector_sock in ready:
+            try:
+                event_meta, event_data = collector_sock.recv_multipart()
+            except ValueError, e:
+                # Sometimes clients can fail and corrupt these two-part sends.
+                log.warning("Failed to recv from %r: %r", collector_sock, e)
+                continue
+
+            try:
+                nonblocking_stream.check_meta_version(event_meta)
+            except ValueError:
+                log.warning("Failed to decode event due to version mismatch")
+                continue
+
+            # See nonblocking_stream.network for how this is packed.
+            _, stream_name, partition_key = struct.unpack(
+                nonblocking_stream.META_STRUCT_FMT, event_meta)
+            waiting_messages[stream_name].append({
+                'Data': event_data,
+                'PartitionKey': partition_key
```

Comment:
My intent is to keep tritond from needing to unpack the full data blob. If all the data for posting the msgpack blob to kinesis is in the meta-data, I don't need to unpack the main data, and can just pass it along packed to kinesis.